timemachine/cpp/src/local_md_potentials.cu

```diff
@@ -199,6 +213,17 @@ void LocalMDPotentials::_setup_free_idxs_given_reference_idx(
     bound_restraint_->set_params_device(std::vector<int>({num_row_idxs, 3}), d_bond_params_.data, stream);
     restraint_->set_bonds_device(num_row_idxs, d_restraint_pairs_.data, stream);
 
+    if (!freeze_reference_) {
+        // Remove the reference idx from the column indices, d_free_idxs gets inverted to construct column idxs,
+        // and add to the row indices
```

Comment:
This is probably pretty inefficient (launching two kernels with 1 thread 1 block).

Would it be simpler to just modify the partition code above (lines 168-170) itself to simplify always include reference in the `free_idxs`?

Reply:
Had originally done it this way to avoid the reference -> reference restraint thatt would be setup. But since r is always 0 in that case, might actually be okay.

Reply:
Yup, works. Though maybe we should have some worry about being able to construct a restraint here that would be invalid to construct from scratch (we verify src_idx != dst_idx).

Added in https://github.com/proteneer/timemachine/pull/1014/commits/37ea591777808702a5ec9f3657b017554fd09b80